src/test/java/bio/terra/service/filedata/azure/util/BlobIOTestUtility.java

```diff
@@ -0,0 +1,157 @@
+
+  public BlobContainerClient getSourceBlobContainerClient() {
+    return sourceBlobContainerClient;
+  }
+
+  public BlobContainerClient getDestinationBlobContainerClient() {
+    return destinationBlobContainerClient;
+  }
+
+  private final BlobContainerClient destinationBlobContainerClient;
+  private String sourceContainerName;
+  private String destinationContainerName;
+  private final String STORAGE_ENDPOINT_PATTERN = "https://%s.blob.core.windows.net/%s";
+  private final TokenCredential tokenCredential;
+
+  public BlobIOTestUtility(
+      TokenCredential tokenCredential, String sourceAccountName, String destinationAccountName) {
+    this.sourceContainerName = generateNewTempContainerName();
+    this.destinationContainerName = generateNewTempContainerName();
+    this.sourceBlobContainerClient =
+        createBlobContainerClient(tokenCredential, this.sourceContainerName, sourceAccountName);
+    this.destinationBlobContainerClient =
+        createBlobContainerClient(
+            tokenCredential, this.destinationContainerName, destinationAccountName);
+    this.sourceBlobContainerClient.create();
+    this.destinationBlobContainerClient.create();
+    this.tokenCredential = tokenCredential;
+  }
+
+  private BlobContainerClient createBlobContainerClient(
+      TokenCredential credential, String containerName, String accountName) {
+
+    return new BlobContainerClientBuilder()
+        .credential(credential)
+        .endpoint(String.format(Locale.ROOT, STORAGE_ENDPOINT_PATTERN, accountName, containerName))
+        .buildClient();
+  }
+
+  public String generateSourceContainerUrlWithSasReadAndListPermissions(String accountKey) {
+    String sasToken = generateContainerSasTokenWithReadAndListPermissions(accountKey);
+
+    return String.format(
+        "%s?%s", this.getSourceBlobContainerClient().getBlobContainerUrl(), sasToken);
+  }
+
+  public String generateContainerSasTokenWithReadAndListPermissions(String accountKey) {
+    BlobSasPermission permissions =
+        new BlobSasPermission().setReadPermission(true).setListPermission(true);
+
+    OffsetDateTime expiryTime = OffsetDateTime.now().plusDays(1);
+    SasProtocol sasProtocol = SasProtocol.HTTPS_ONLY;
+
+    // build the token
+    BlobServiceSasSignatureValues sasSignatureValues =
+        new BlobServiceSasSignatureValues(expiryTime, permissions).setProtocol(sasProtocol);
+
+    BlobContainerClient blobContainerClient =
+        new BlobContainerClientBuilder()
+            .credential(
+                new StorageSharedKeyCredential(
+                    this.sourceBlobContainerClient.getAccountName(), accountKey))
+            .endpoint(this.sourceBlobContainerClient.getBlobContainerUrl())
+            .buildClient();
+
+    return blobContainerClient.generateSas(sasSignatureValues);
+  }
+
+  private String generateNewTempContainerName() {
+    return UUID.randomUUID().toString().toLowerCase().replace("-", "").substring(0, 30);
```

Comment:
I have a method I created to do this that we should use here:
```AzureDataLocationSelector.armUniqueString(RandomStringUtils.random(10), 30)```

Reply:
Hmmm I don't it will work. RamdomStringUtils could return invalid chars from the perspective of container naming rules .

Reply:
Right, that's just to have a random seed.  `AzureDataLocationSelector.armUniqueString` generates a valid container name

Reply:
will change